### PR TITLE
Allow user defined potentials for hmc

### PR DESCRIPTION
**Pull request by aseyboldt**
pymc3 should pass user defined potentials to hmc. This used to be in #1747.

### CODE REVIEW

File: pymc3/step_methods/hmc/base_hmc.py [hunk 1]
```diff
@@ -38,15 +42,18 @@ def __init__(self, vars=None, scaling=None, step_scale=0.25, is_cov=False,
...
             scaling = guess_scaling(Point(scaling, model=model), model=model, vars=vars)
 
-        n = scaling.shape[0]
+        if scaling is not None and potential is not None:
+            raise ValueError("Can not specify both potential and scaling.")
+
+        n = model.dict_to_array(model.test_point).shape[0]
         self.step_size = step_scale / (n ** 0.25)
-        self.potential = quad_potential(scaling, is_cov, as_cov=False)
+        self.potential = quad_potential(scaling or potential, is_cov, as_cov=False)
```
Comment (twiecki) [inline pymc3/step_methods/hmc/base_hmc.py]: I think this should be outside, `quad_potential()`, no? Also, I think better to be explicit here, i.e. :`if potential is not None: self.potential = potential`.

File: pymc3/step_methods/hmc/quadpotential.py [hunk 2]
```diff
@@ -29,6 +30,8 @@ def quad_potential(C, is_cov, as_cov):
     -------
     q : Quadpotential
     """
+    if isquadpotential(C):
```
Comment (twiecki) [inline pymc3/step_methods/hmc/quadpotential.py]: Wouldn't need this then.

File: pymc3/step_methods/hmc/base_hmc.py (hunk 1 above)
Comment (aseyboldt) [reply pymc3/step_methods/hmc/base_hmc.py]: true

**Comment by twiecki** [comment]
LGTM.

**Pull request merged**